build_tools/cmake/iree_configure_testing.cmake

```diff
@@ -0,0 +1,79 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Functions for setting up testing in the IREE project. Configures some specific
+# environment variables and sets up the creation of test-specific temporary
+# directories.
+
+enable_testing(iree)
+# A property is apparently the only way to get an uncached global variable.
+set_property(GLOBAL PROPERTY IREE_TEST_TMPDIRS "")
+set(IREE_TEST_TMPDIR_ROOT "${IREE_BINARY_DIR}/test_tmpdir")
+
+# iree_configure_test
+#
+# Registers test for temporary directory creation and adds properties common to
+# all IREE tests. This should be invoked with each test added with `add_test`.
+#
+# Parameters:
+#   TEST_NAME: the test name, e.g. iree/base/math_test
+function(iree_configure_test TEST_NAME)
+  set(_TEST_TMPDIR "${IREE_TEST_TMPDIR_ROOT}/${TEST_NAME}_test_tmpdir")
+  set_property(GLOBAL APPEND PROPERTY IREE_TEST_TMPDIRS ${_TEST_TMPDIR})
+  set_property(TEST ${TEST_NAME} APPEND PROPERTY ENVIRONMENT "TEST_TMPDIR=${_TEST_TMPDIR}")
+
+  # IREE_*_DISABLE environment variables may used to skip test cases which
+  # require both a compiler target backend and compatible runtime HAL driver.
+  #
+  # These variables may be set by the test environment, typically as a property
+  # of some continuous execution test runner or by an individual developer, or
+  # here by the build system.
+  #
+  # Tests which only depend on a compiler target backend or a runtime HAL
+  # driver, but not both, should generally use a different method of filtering.
+  if(NOT IREE_TARGET_BACKEND_VULKAN_SPIRV OR NOT IREE_HAL_DRIVER_VULKAN)
+    set_property(TEST ${TEST_NAME} APPEND PROPERTY ENVIRONMENT "IREE_VULKAN_DISABLE=1")
+  endif()
```

Comment:
FYI I had my eye on some of this when bringing up the web platform tests. The https://github.com/google/iree/blob/main/experimental/web/testing/parse_test_list.py script currently just ignores all `ENVIRONMENT` properties, but I might want to either omit the environment properties or handle them in some web-specific way.